parser/types/field_type.go

```diff
@@ -64,7 +65,12 @@ func NewFieldType(tp byte) *FieldType {
 		decimal: UnspecifiedLength,
 	}
 }
-
+func (ft *FieldType) IsDecimalValid() bool {
+	if ft.tp == mysql.TypeNewDecimal && (ft.decimal < 0 || ft.decimal > mysql.MaxDecimalScale || ft.flen <= 0 || ft.flen > mysql.MaxDecimalWidth || ft.flen <= ft.decimal) {
```

Comment:
Just quesction: is `ft.flen <= ft.decimal` reasonable? Because SQL like `create table t1(c1 decimal(10, 10));` is correct. Rest LGTM